osu.Game/Online/API/APIMod.cs

```diff
@@ -75,6 +76,16 @@ public bool Equals(APIMod other)
             return Acronym == other.Acronym && Settings.SequenceEqual(other.Settings, ModSettingsEqualityComparer.Default);
         }
 
+        [SuppressMessage("ReSharper", "NonReadonlyMemberInGetHashCode")]
+        public override int GetHashCode()
+        {
+            var hashCode = new HashCode();
+            hashCode.Add(Acronym);
+            foreach (var s in Settings)
+                hashCode.Add(ModSettingsEqualityComparer.Default.GetHashCode(s));
```

Comment:
Because dictionary enumeration order is not explicitly defined, this might decrease hit rate when looking up mods with custom setting values applied. A way to improve that would be to sort here (by key, most likely) before adding to the `HashCode`.

Unsure if worth in general, because I'm suspecting that cache hit rate for difficulty values for mods with custom settings applied will be very low (especially with something like difficulty adjust), and so I'd probably consider *not* caching anything *at all* if there are custom settings applied, until it becomes a problem at least. But that's maybe a bit out of scope for this pull.

Reply:
I'm still not sure we should ever be storing something like `APIMod` as a key in a dictionary, as it has too much variation (as came up in discussion yesterday with @smoogipoo). ie. there could be unlimited combinations of mod settings, to the point it may not make sense to require this in the first place..